### PR TITLE
unavailable image asset fix and available bands refresh button

### LINKED ISSUE #1189: Address GeoDash issues found during testing
- [x] Issues found by Karen
- [x] Issues found by Roberto

### PR DESCRIPTION
**Pull request by rfontanarosa**
fix #1189

Available Bands was an hidden row, now it is always visible.

GeoDash didn't know the difference between an Image Asset or an Image Collection Asset, this caused errors when it requested for the Available Bands.

Instead of performing an http request to GEEG at onChange time, now the user have to click on the Refresh button to get the available bands for a specific Image Asset or Image Collection Asset.

### CODE REVIEW

**Comment by sirmspencer** [comment]
src/js/widgetLayoutEditor.js
288-291, Dont mutate postObjects
```
const postObjects = {
	path: "getAvailableBands",
	...(isCollection ? {imageCollection: value} : {image: value})
}

1292, 1300, these are rendering the same thing.  Just pass in availableBands / availableBandsDual depending on where its used.

**Comment by rjsheperd** [comment]
I noticed that our linter has been complaining about the geodash file you edited for this PR. If you wouldn't mind:

* updating the `eval` statements to `JSON.parse` and testing that GeoDash is still working properly, 
* Running `npm i -S react-resizable` and adding the `package.json` file to your PR

File: src/js/widgetLayoutEditor.js [hunk 1]
```diff
@@ -282,12 +283,12 @@ class WidgetLayoutEditor extends React.PureComponent {
...
-    getBandsFromGateway = isDual => {
-        // go get available bands
-        if (event.target.value !== "custom") {
+    getBandsFromGateway = (isDual, isCollection) => {
+        const value = isDual ? this.state.imageCollectionDual : this.state.imageCollection;
+        if (value !== "") {
             const postObject = {
                 path: "getAvailableBands",
-                imageCollection: event.target.value // "LANDSAT/LT05/C01/T1"
+                ...(isCollection ? {imageCollection: value} : {image: value}),
```
Comment (rjsheperd) [inline src/js/widgetLayoutEditor.js:291]: Last linter warning: `Unexpected trailing comma`

**Review by sirmspencer — CHANGES_REQUESTED** [review]
While I appreciate your effort to eliminate the assignment warnings, it appears we need to go over how to create immutable functions involving objects.

**Review by sirmspencer — CHANGES_REQUESTED** [review]
src/js/widgetLayoutEditor.js
1378, 1547, 1585 You can condense this logic.  Note that each line refers to a different state (selectedWidgetType, selectedDataType, selectedDataTypeDual)
```
const [label, placeholder, isCollection] = this.state.selectedWidgetType === "imageAsset"
    ? ["GEE Image Asset", "LANDSAT/LC8_L1T_TOA/LC81290502015036LGN00", false]
    : ["GEE Collection Image Asset", "LANDSAT/LC8_L1T_TOA", true];
```
1544 move `||` to the beginning of the next line. Its easier to find operators when they are in the front.

**Pull request merged**